cross-region-replication-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationConverter.java

```diff
@@ -0,0 +1,103 @@
+    private GlueSchemaRegistryDeserializerImpl deserializer;
+    private GlueSchemaRegistrySerializerImpl serializer;
+    private boolean isKey;
+
+    /**
+     * Constructor used by Kafka Connect user.
+     */
+    public CrossRegionReplicationConverter(){};
+
+    /**
+     * Constructor accepting AWSCredentialsProvider.
+     *
+     * @param credentialsProvider AWSCredentialsProvider instance.
+     */
+    public CrossRegionReplicationConverter(
+            AwsCredentialsProvider credentialsProvider,
+            GlueSchemaRegistryDeserializerImpl deserializerImpl,
+            GlueSchemaRegistrySerializerImpl serializerImpl) {
+
+        this.credentialsProvider = credentialsProvider;
+        this.deserializer = deserializerImpl;
+        this.serializer = serializerImpl;
+    }
+
+    /**
+     * Configure the Schema Replication Converter.
+     * @param configs configuration elements for the converter
+     * @param isKey true if key, false otherwise
+     */
+    @Override
+    public void configure(Map<String, ?> configs, boolean isKey) {
+        this.isKey = isKey;
+        //TODO: Support credentialProvider passed on by the user
+        credentialsProvider = DefaultCredentialsProvider.builder().build();
+
+        // Put the source and target regions into configurations respectively
+        Map<String, Object> sourceConfigs = new HashMap<>(configs);
+        Map<String, Object> targetConfigs = new HashMap<>(configs);
+
+        if (configs.get(AWSSchemaRegistryConstants.AWS_SOURCE_REGION) == null){
+            throw new DataException("Source Region is not provided.");
+        }
+        else if (configs.get(AWSSchemaRegistryConstants.AWS_TARGET_REGION) == null && configs.get(AWSSchemaRegistryConstants.AWS_REGION) == null){
+            throw new DataException("Target Region is not provided.");
+        }
+
+        sourceConfigs.put(AWSSchemaRegistryConstants.AWS_REGION, configs.get(AWSSchemaRegistryConstants.AWS_SOURCE_REGION));
+        targetConfigs.put(AWSSchemaRegistryConstants.AWS_REGION, configs.get(AWSSchemaRegistryConstants.AWS_TARGET_REGION));
+
+        serializer   = new GlueSchemaRegistrySerializerImpl(credentialsProvider, new GlueSchemaRegistryConfiguration(targetConfigs));
+        deserializer = new GlueSchemaRegistryDeserializerImpl(credentialsProvider, new GlueSchemaRegistryConfiguration(sourceConfigs));
+    }
+
+    @Override
+    public byte[] fromConnectData(String topic, org.apache.kafka.connect.data.Schema schema, Object value) {
+        if (value == null) return null;
+        byte[] bytes = (byte[]) value;
+
+        try {
+            byte[] deserializedBytes = deserializer.getData(bytes);
+            Schema deserializedSchema = deserializer.getSchema(bytes);
+            //The registry is decided by the configuration in the target region , schema name is the same as the source region
+            return serializer.encode(topic, deserializedSchema, deserializedBytes);
```

Comment:
are you going to work on prefixing topic name the source cluster alias so we can have the registered schema name same as topic name in target region?

Reply:
If you will not work on it then please also create a issue in the backlog for us to keep track. Link the issue in TODO as well.

Reply:
Done!